manticore/core/state.py

```diff
@@ -240,6 +241,40 @@ def record_branches(self, targets):
             except KeyError:
                 self.branches[key] = 1
 
+    def generate_inputs(self, workspace, generate_files = False):
+        '''
+        Save the inputs of the state
+
+        :param str workspace: the working directory 
+        :param bool generate_files: true if symbolic files are also generated
+        '''
+
+        # Save constraints formula
+        smtfile = 'state_{:08x}.smt'.format(self.co)
+        with open(os.path.join(workspace,smtfile), 'wb') as f:
+            f.write(str(self.constraints))
+
+        # check that the state is sat
+        assert solver.check(self.constraints)
+
+        # save the inputs
+        for symbol in self.input_symbols:
+            buf = solver.get_value(self.constraints, symbol)
+            filename = os.path.join(workspace,'state_%08x.txt'%self.co)
+            file(filename,'a').write("%s: %s\n"%(symbol.name, repr(buf)))
```

Comment:
open() is preferred. Consider using it?
https://slack-files.com/T03A74C1E-F5H2XCNCU-7d40a2b8eb